python/ray/scripts/scripts.py

```diff
@@ -827,6 +827,13 @@ def stop(force):
             corpus = (proc_cmd
                       if filter_by_cmd else subprocess.list2cmdline(proc_args))
             if keyword in corpus:
+                # This is a way to avoid killing redis server that's not started by Ray.
+                # We are using a simple hacky solution here since
+                # Redis server will anyway removed soon from the ray repository.
+                if (keyword == "redis-server"
+                        and "core/src/ray/thirdparty/redis/src/redis-server"
```

Comment:
this looks a bit fragile too... i.e. it probably won't work with windows, and subject to failure if the path changes.